lib/rules/render-result-naming-convention.ts

```diff
@@ -0,0 +1,101 @@
+import { ESLintUtils, TSESTree } from '@typescript-eslint/experimental-utils';
+import { getDocsUrl, hasTestingLibraryImportModule } from '../utils';
+import {
+  isCallExpression,
+  isIdentifier,
+  isImportSpecifier,
+  isObjectPattern,
+  isRenderVariableDeclarator,
+} from '../node-utils';
+
+export const RULE_NAME = 'render-result-naming-convention';
+
+const ALLOWED_VAR_NAMES = ['view', 'utils'];
+const ALLOWED_VAR_NAMES_TEXT = ALLOWED_VAR_NAMES.map(
+  name => '`' + name + '`'
+).join(', ');
+
+export default ESLintUtils.RuleCreator(getDocsUrl)({
+  name: RULE_NAME,
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'TODO',
+      category: 'Best Practices',
+      recommended: false,
+    },
+    messages: {
+      invalidRenderResultName: `\`{{ varName }}\` is not a recommended name for \`render\` returned value. Instead, you should destructure it, or call it using one of the valid choices: ${ALLOWED_VAR_NAMES_TEXT}`,
+    },
+    fixable: null,
+    schema: [
+      {
+        type: 'object',
+        properties: {
+          renderFunctions: {
+            type: 'array',
+          },
+        },
+      },
+    ],
+  },
+  defaultOptions: [
+    {
+      renderFunctions: [],
+    },
+  ],
+
+  create(context, [options]) {
+    const { renderFunctions } = options;
+    let renderResultName: string | null = null;
+    let renderAlias: string | undefined;
+
+    return {
+      ImportDeclaration(node: TSESTree.ImportDeclaration) {
```

Comment:
should we also consider when importing in the following format?

```ts
import * as dom from '@testing-library/react'
```

Some of the rules check that.

Reply:
Ah of course! I forgot about that variant. I'll add this later.